lib/whitehall_importer/integrity_checker/change_history_check.rb

```diff
@@ -0,0 +1,44 @@
+module WhitehallImporter
+  class IntegrityChecker::ChangeHistoryCheck
+    attr_reader :edition, :proposed_change_history, :publishing_api_change_history
+
+    def initialize(edition, proposed_change_history, publishing_api_change_history)
+      @edition = edition
+      @proposed_change_history = proposed_change_history
+      @publishing_api_change_history = publishing_api_change_history
+    end
+
+    def match?
+      return false unless history_length_matches?
+
+      if edition.draft?
+        history_excluding_first_timestamp_matches?
+      else
+        history_matches?(proposed_change_history, publishing_api_change_history)
+      end
+    end
+
+  private
+
+    def history_length_matches?
+      proposed_change_history.length == publishing_api_change_history.length
+    end
+
+    def history_matches?(proposed, publishing_api)
+      proposed.zip(publishing_api).all? do |proposed_history, publishing_api_history|
+        proposed_history[:note] == publishing_api_history[:note] &&
+          proposed_history[:public_timestamp] == publishing_api_history[:public_timestamp]
```

Comment:
I reckon we'll want to use https://github.com/alphagov/content-publisher/pull/1875/files#diff-93e7bf043d189b6ad9f28ef07d33176bR5 for this once it's merged.

Reply:
@Rosa-Fox This has been merged now so we can change to use it `IntegrityChecker.time_matches?(...)`